### PR TITLE
Remove index and parentId from GroupTree node data

### PR DESCRIPTION
**Pull request by hugo-vrijswijk**
The GroupTree node case classes `GroupObs`/`Grouping` had some duplicate data in `parentId`, `parentIndex` and `elements`. Since this data is already known by the tree, we can remove it from the case classes and only use the node data for this.

- Also remove intermediate ObsNode class that was used for ObsList (this greatly simplifies the Iso used!)
- Use `NonNegInt` for KeyedIndexed(Tree|List)

### CODE REVIEW

File: model/shared/src/main/scala/explore/data/KeyedIndexedList.scala [hunk 1]
```diff
@@ -139,7 +145,7 @@ object KeyedIndexedList:
               .map(list =>
                 KeyedIndexedList.unsafeFromTreeSeqMap(
                   TreeSeqMap.from(
-                    list.sortBy { case (_, (_, i)) => i }
+                    list.sortBy { case (_, (_, i)) => i }(Order[NonNegInt].toOrdering)
```
Comment (hugo-vrijswijk) [inline model/shared/src/main/scala/explore/data/KeyedIndexedList.scala:148]: I'm not sure why, but I wasn't able to derive an `Ordering[NonNegInt]` here, so instead I summon the `Order[NonNegInt]` and convert that. If I'm missing an import then I'd be happy to replace it

Comment (rpiaggio) [reply model/shared/src/main/scala/explore/data/KeyedIndexedList.scala:148]: Usually, if you `import cats.Order.given`, it derives an `Ordering` from an implicit `Order` in scope. Not sure whether we can bring the `Order` given into scope here though.

File: explore/src/main/scala/explore/tabs/GroupEditTile.scala [hunk 2]
```diff
@@ -242,6 +248,6 @@ object GroupEditTile:
     val andOrStr: VdomNode =
       if group.isAnd then if group.ordered then "Ordered" else "Any order"
       else
-        s"Choose ${group.minimumRequired.getOrElse(NonNegShort.unsafeFrom(1))} of ${group.elements.length}"
+        s"Choose ${group.minimumRequired.getOrElse(NonNegShort.unsafeFrom(1))} of ${elementsLength}"
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/tabs/GroupEditTile.scala:251]: `1.refined` ?

File: model/shared/src/main/scala/explore/data/KeyedIndexedList.scala (hunk 1 above)
Comment (hugo-vrijswijk) [reply model/shared/src/main/scala/explore/data/KeyedIndexedList.scala]: Yup, that fixed it

File: explore/src/main/scala/explore/tabs/GroupEditTile.scala (hunk 2 above)
Comment (hugo-vrijswijk) [reply explore/src/main/scala/explore/tabs/GroupEditTile.scala]: Yep

**Pull request merged**